cvat-ui/src/actions/export-actions.ts

```diff
@@ -61,68 +71,105 @@ export const exportActions = {
     closeExportBackupModal: (instance: any) => (
         createAction(ExportActionTypes.CLOSE_EXPORT_BACKUP_MODAL, { instance })
     ),
-    exportBackup: (instance: any) => (
+    exportBackup: (instance: Exclude<InstanceType, Job> | RequestInstanceType) => (
         createAction(ExportActionTypes.EXPORT_BACKUP, { instance })
     ),
-    exportBackupSuccess: (instance: any, instanceType: 'task' | 'project', isLocal: boolean) => (
-        createAction(ExportActionTypes.EXPORT_BACKUP_SUCCESS, { instance, instanceType, isLocal })
+    exportBackupSuccess: (instance: Exclude<InstanceType, Job> | RequestInstanceType, instanceType: 'task' | 'project', target: 'local' | 'cloudstorage' | undefined) => (
+        createAction(ExportActionTypes.EXPORT_BACKUP_SUCCESS, { instance, instanceType, target })
     ),
-    exportBackupFailed: (instance: any, instanceType: 'task' | 'project', error: any) => (
+    exportBackupFailed: (instance: Exclude<InstanceType, Job> | RequestInstanceType, instanceType: 'task' | 'project', error: any) => (
         createAction(ExportActionTypes.EXPORT_BACKUP_FAILED, { instance, instanceType, error })
     ),
 };
 
+export async function listenExportDatasetAsync(
+    rqID: string,
+    dispatch: (action: ExportActions | RequestsActions) => void,
+    params: {
+        instance: InstanceType | RequestInstanceType,
+        format: string,
+        saveImages: boolean,
+    },
+): Promise<void> {
+    const { instance, format, saveImages } = params;
+    const resource = saveImages ? 'dataset' : 'annotations';
+
+    const instanceType = getInstanceType(instance);
+    try {
+        let result;
+        let target: 'cloudstorage' | 'local' | undefined;
+        if (rqID) {
```

Comment:
Declared type is string, but we have this condition here

Reply:
Added `undefined`